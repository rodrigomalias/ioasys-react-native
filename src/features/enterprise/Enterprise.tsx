import * as React from "react"
import { Image, ScrollView, StyleSheet, Text, View } from "react-native"
import { hasObject, translation, firstLetter } from "../../utils"
import { IEnterpriseReducer } from "../../store/modules/enterprise/State"
import { RouteProp, useRoute } from "@react-navigation/native"
import LoadingSpinner from "../../components/LoadingSpinner"
import { colors } from "../../colors"

export const Enterprise = (props: IEnterpriseReducer) => {
    const { getEnterpriseById, enterprise, errorEnterprise } = props

    const [ loading, setLoading ] = React.useState(true)

    const route: RouteProp<{ params: { enterpriseId: number } }, "params"> = useRoute()
    const { enterpriseId } = route.params

    const imageUrl = `https://empresas.ioasys.com.br/${enterprise.photo}`

    React.useEffect(() => {
        getEnterpriseById({ enterpriseId })
    }, [ enterpriseId, getEnterpriseById, route.params ])

    React.useEffect(() => {
        if (hasObject(errorEnterprise) || hasObject(enterprise) && (enterprise.id == enterpriseId)) {
            setLoading(false)
        }
    }, [ enterprise, enterpriseId, errorEnterprise ])

    return (
        <View style={styles.container}>
            <ScrollView>
                <View style={styles.contentContainer}>
                    {loading ? (
                        <LoadingSpinner loading={loading}/>
                    ) : (
                        <View style={styles.enterpriseContainer}>
                            <Image source={{ uri: imageUrl }} style={styles.image} />
                            <Text style={styles.enterpriseDescription}>
                                {enterprise.description}
                            </Text>
                            <Text style={styles.title}>
                                {firstLetter(translation("address"))}
                            </Text>
                            <Text>
                                {enterprise.country} {enterprise.city}
                            </Text>
                        </View>
                    )}
                </View>
            </ScrollView>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.whiteBackground,
    },
    contentContainer: {
        paddingTop: 50,
        paddingBottom: 50,
    },
    enterpriseContainer: {
        alignItems: "center",
    },
    title: {
        textAlign: "justify",
        margin: 12,
        fontSize: 18,
        fontWeight: "bold",
    },
    image: {
        width: 150,
        height: 150,
        borderRadius: 150,
        borderWidth: 1,
        borderColor: colors.lightGray,
        shadowColor: colors.black,
        shadowOffset: {
            width: 2,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
    },
    enterpriseDescription: {
        textAlign: "justify",
        margin: 20,
        color: colors.darkGray,
        lineHeight: 24,
    }
})