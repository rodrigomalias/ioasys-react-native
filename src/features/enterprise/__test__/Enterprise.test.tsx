import React from "react"
import { Enterprise } from "../Enterprise"
import { render } from "@testing-library/react-native"
import renderer from "react-test-renderer"

const mockValue = {
    city: "SP",
    country: "Brazil",
    description: "Description test",
    email_enterprise: "email@email.com",
    enterprise_name: "Company test",
    enterprise_type: {
        enterprise_type_name: "Type test",
        id: 1
    },
    facebook: undefined,
    id: 1,
    linkedin: undefined,
    own_enterprise: false,
    phone: undefined,
    photo: "photo.png",
    share_price: 5000,
    shares: undefined,
    twitter: undefined,
    value: 500
}

const initialState = {
    getEnterpriseById: () => jest.fn(),
    getEnterprisesNameOrType: () => jest.fn(),
    getEnterprises: () => jest.fn(),
    enterprise: mockValue,
    enterprises: [ mockValue ],
    errorEnterprise: "",
    errorEnterprises: "",
}

describe("<Enterprise/>", () => {
    test("should render correctly", () => {
        const tree = renderer.create(<Enterprise {...initialState}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test("should find the image using testId", () => {
        const { getByTestId } = render(<Enterprise {...initialState} />)

        const image = getByTestId("test_enterprise_image")

        expect(image).not.toBeNull()
    })

    test("should find the scrollView using testId", () => {
        const testIdName = "test_enterprise_scrollview"

        const { getByTestId } = render(<Enterprise {...initialState} />)

        const foundScrollView = getByTestId(testIdName)

        expect(foundScrollView).toBeTruthy()
    })
})