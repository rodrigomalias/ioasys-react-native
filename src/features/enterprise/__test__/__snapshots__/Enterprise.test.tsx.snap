// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Enterprise/> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    testID="test_enterprise_scrollview"
  >
    <View>
      <View
        style={
          Object {
            "paddingBottom": 50,
            "paddingTop": 50,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://empresas.ioasys.com.br/photo.png",
              }
            }
            style={
              Object {
                "borderColor": "#C0C0C0",
                "borderRadius": 150,
                "borderWidth": 1,
                "height": 150,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 150,
              }
            }
            testID="test_enterprise_image"
          />
          <Text
            style={
              Object {
                "color": "#999",
                "lineHeight": 24,
                "margin": 20,
                "textAlign": "justify",
              }
            }
          >
            Description test
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "margin": 12,
                "textAlign": "justify",
              }
            }
          >
            Endere√ßo
          </Text>
          <Text>
            Brazil
             
            SP
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
